mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -100,16 +101,20 @@ where
 
         let result_coin = (self.constructor)(utxo_arc.clone());
         if let Some(sync_status_loop_handle) = sync_status_loop_handle {
-            let current_block_height = result_coin
-                .current_block()
-                .compat()
-                .await
-                .map_to_mm(UtxoCoinBuildError::GetCurrentBlockHeightError)?;
+            let block_count = match result_coin.as_ref().rpc_client.get_block_count().compat().await {
```

Comment:
I think we can fetch the block count inside of `block_header_utxo_loop` function, in particular, before the loop.

Reply:
I guess this isn't needed as we discussed since both are basically still same operation and even this will require passing more args(larger) than required to `block_header_utxo_loop`